src/output-json-anomaly.c

```diff
@@ -349,9 +363,19 @@ static void SetFlag(const ConfNode *conf, const char *name, uint16_t flag, uint1
 static void JsonAnomalyLogConf(AnomalyJsonOutputCtx *json_output_ctx,
         ConfNode *conf)
 {
+    static bool _warn_no_flags = false;
     uint16_t flags = 0;
     if (conf != NULL) {
-        SetFlag(conf, "packethdr", LOG_JSON_PACKETHDR, &flags);
+        SetFlag(conf, "protoparser", LOG_JSON_PROTOPARSER, &flags);
+        SetFlag(conf, "parser",      LOG_JSON_PARSER,      &flags);
+        SetFlag(conf, "protodetect", LOG_JSON_PROTODETECT, &flags);
+        SetFlag(conf, "protodecode", LOG_JSON_PROTODECODE, &flags);
+        SetFlag(conf, "packethdr",   LOG_JSON_PACKETHDR,   &flags);
+    }
+    if (flags == 0 && !_warn_no_flags) {
+        SCLogNotice("Anomaly logging has been configured; however, no logging filters "
+                    "have been selected. Select one or more protocol/parser filters.");
+        _warn_no_flags = true;
```

Comment:
this will only display the message once even if we have multiple eve-log instances with anomaly enabled, right?

Reply:
Correct.